File: DataInjectorAndroid/DataInjectorLib/src/main/java/com/crescentflare/datainjector/condition/InjectorConditionType.java
package com.crescentflare.datainjector.condition;

/**
 * Data injector condition: condition type enum
 * Determines the type of the condition item in a sequence
 */
public enum InjectorConditionType
{
    // ---
    // Enum values
    // ---

    Unknown(""),
    Equals("=="),
    NotEquals("!="),
    Bigger(">"),
    Smaller("<"),
    BiggerOrEquals(">="),
    SmallerOrEquals("<="),
    And("&&"),
    Or("||");


    // ---
    // Link string to enum
    // ---

    private String string;

    InjectorConditionType(String string)
    {
        this.string = string;
    }

    public String toString()
    {
        return this.string;
    }

    public static InjectorConditionType fromString(String string)
    {
        for (InjectorConditionType type : values())
        {
            if (type.string.length() > 0 && type.string.equals(string))
            {
                return type;
            }
        }
        return Unknown;
    }


    // ---
    // Helpers
    // ---

    public boolean isComparison()
    {
        return this == Equals || this == NotEquals || this == Bigger || this == Smaller || this == BiggerOrEquals || this == SmallerOrEquals;
    }

    public boolean isOperator()
    {
        return this == And || this == Or;
    }

    public static boolean isReservedCharacter(char chr)
    {
        for (InjectorConditionType type : values())
        {
            if (type.string.length() > 0 && type.string.charAt(0) == chr)
            {
                return true;
            }
        }
        return false;
    }
}

File: DataInjectorIOS/DataInjector/Classes/condition/InjectorConditionType.swift
//
//  InjectorConditionType.swift
//  DataInjector Pod
//
//  Library condition: condition type enum
//  Determines the type of the condition item in a sequence
//

import UIKit

/// A type of condition element used in the injector condition class
enum InjectorConditionType: String {

    // ---
    // MARK: Enum values
    // ---

    case unknown = ""
    case equals = "=="
    case notEquals = "!="
    case bigger = ">"
    case smaller = "<"
    case biggerOrEquals = ">="
    case smallerOrEquals = "<="
    case and = "&&"
    case or = "||"
    
    static func allValues() -> [InjectorConditionType] {
        return [equals, notEquals, bigger, smaller, biggerOrEquals, smallerOrEquals, and, or]
    }
    

    // ---
    // MARK: Helpers
    // ---
    
    func isComparison() -> Bool {
        return self == .equals || self == .notEquals || self == .bigger || self == .smaller || self == .biggerOrEquals || self == .smallerOrEquals
    }
    
    func isOperator() -> Bool {
        return self == .and || self == .or
    }

    static func isReservedCharacter(chr: Character) -> Bool {
        for value in allValues() {
            if value.rawValue.characters.count > 0 {
                let firstChr = value.rawValue.characters[value.rawValue.startIndex]
                if firstChr == chr {
                    return true
                }
            }
        }
        return false
    }

}
